libnmstate/nm/device.py

```diff
@@ -22,15 +22,12 @@
 from libnmstate.error import NmstateLibnmError
 
 from . import active_connection as ac
-from . import connection
+from . import profile
 
 
-def activate(context, dev=None, connection_id=None):
-    """Activate the given device or remote connection profile."""
-    conn = connection.ConnectionProfile(context)
-    conn.nmdevice = dev
-    conn.con_id = connection_id
-    conn.activate()
+def activate(context, dev=None, profile=None):
+    """Activate the given remote connection profile."""
+    profile.activate()
```

Comment:
If this function is used by test only, please remove it and find a better way for test usage.

The same for other `nm.device` functions.